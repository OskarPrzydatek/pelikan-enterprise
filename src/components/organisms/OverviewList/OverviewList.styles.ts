import styled, { css } from 'styled-components';

export const OverviewList = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const OverviewListWrapper = styled.ul`
  width: 100%;
  height: 70%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Pagination = styled.div`
  margin: ${({ theme }) => theme.spaces.l} 0;
`;

export const navigateButtonCSS = css`
  width: 80%;

  @media ${({ theme }) => theme.breakpoints.tablet} {
    width: 20%;
  }
`;
