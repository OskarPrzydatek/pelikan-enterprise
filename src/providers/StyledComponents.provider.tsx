import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
