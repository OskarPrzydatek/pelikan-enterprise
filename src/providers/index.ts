export { StyledComponentsProvider } from './StyledComponents.provider';
