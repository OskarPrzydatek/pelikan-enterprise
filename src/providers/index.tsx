import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
