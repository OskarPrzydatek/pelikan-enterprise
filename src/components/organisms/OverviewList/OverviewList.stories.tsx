import { Meta, StoryObj } from '@storybook/react';

import { StyledComponentsProvider } from '~/providers';

import { OverviewList } from './OverviewList';

const meta: Meta<typeof OverviewList> = {
  title: 'Organisms/OverviewList',
  component: OverviewList,
};

export default meta;
type Story = StoryObj<typeof OverviewList>;

export const OverviewListWithoutItems: Story = {
  render: () => (
    <StyledComponentsProvider>
      <OverviewList
        items={[]}
        navigateLabel="Dodaj atrakcję"
        noItemsLabel="Brak atrakcji w systemie"
        page="1"
        title="Atrakcje"
        onClickItem={() => console.log('onClickItem')}
        onClickNavigate={() => console.log('onClickNavigate')}
      />
    </StyledComponentsProvider>
  ),
};

export const OverviewListWithItems: Story = {
  render: () => (
    <StyledComponentsProvider>
      <OverviewList
        navigateLabel="Dodaj atrakcję"
        noItemsLabel="Brak atrakcji w systemie"
        page="1"
        title="Atrakcje"
        items={[
          { id: 1, name: 'attraction1' },
          { id: 2, name: 'attraction2' },
          { id: 3, name: 'attraction3' },
          { id: 4, name: 'attraction4' },
          { id: 5, name: 'attraction5' },
        ]}
        onClickItem={() => console.log('onClickItem')}
        onClickNavigate={() => console.log('onClickNavigate')}
      />
    </StyledComponentsProvider>
  ),
};
