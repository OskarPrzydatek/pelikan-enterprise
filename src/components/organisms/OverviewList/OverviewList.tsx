import React from 'react';

import { Button, Header, Icon, Text } from '~/components/atoms';
import { Icons } from '~/constants';

import * as S from './OverviewList.styles';

interface IOverviewList {
  title: string;
  noItemsLabel: string;
  navigateLabel: string;
  numberOfPages: number;
  children?: React.ReactNode;
  page?: number;
  onClickNextPage: () => void;
  onClickPrevPage: () => void;
  onClickNavigateToCreatePage?: () => void;
}

export const OverviewList: React.FC<IOverviewList> = ({
  title,
  noItemsLabel,
  navigateLabel,
  numberOfPages,
  children,
  page,
  onClickNextPage,
  onClickPrevPage,
  onClickNavigateToCreatePage,
}: IOverviewList) => {
  const itemsLength = React.Children.count(children);

  return (
    <S.OverviewListWrapper>
      <Header>{title}</Header>
      <S.OverviewList>
        {itemsLength > 0 ? (
          children
        ) : (
          <li data-testid="no-items-label">
            <Text>{noItemsLabel}</Text>
          </li>
        )}
      </S.OverviewList>
      <S.Pagination>
        <Icon
          css={S.prevArrowCSS}
          icon={Icons.ARROW}
          onClick={onClickPrevPage}
        />
        <Text dataTestID="pagination-counter">
          {page} / {numberOfPages}
        </Text>
        <Icon
          css={S.nextArrowCSS}
          icon={Icons.ARROW}
          onClick={onClickNextPage}
        />
      </S.Pagination>
      {onClickNavigateToCreatePage ? (
        <Button
          css={S.navigateButtonCSS}
          dataTestID="overview-list-navigate-to-create-page"
          onClick={onClickNavigateToCreatePage}
        >
          {navigateLabel}
        </Button>
      ) : null}
    </S.OverviewListWrapper>
  );
};
