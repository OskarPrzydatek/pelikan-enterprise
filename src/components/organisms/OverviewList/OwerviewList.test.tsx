import { render, screen } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';

import { StyledComponentsProvider } from '~/providers';

import { OverviewList } from './OverviewList';

const mockOnClickNavigate = vi.fn();

describe('OverviewList', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <OverviewList
          navigateLabel="Dodaj atrakcję"
          noItemsLabel="Brak atrakcji w systemie"
          page="1"
          title="Atrakcje"
          onClickNavigate={mockOnClickNavigate}
        />
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });

  test('ensure current page is one when is undefined', () => {
    const { rerender } = render(
      <StyledComponentsProvider>
        <OverviewList
          navigateLabel="Dodaj atrakcję"
          noItemsLabel="Brak atrakcji w systemie"
          page={undefined}
          title="Atrakcje"
          onClickNavigate={mockOnClickNavigate}
        />
      </StyledComponentsProvider>
    );
    const paginationCounter = screen.getByTestId('pagination-counter');
    expect(paginationCounter).toHaveTextContent('1 / X');
    rerender(
      <StyledComponentsProvider>
        <OverviewList
          navigateLabel="Dodaj atrakcję"
          noItemsLabel="Brak atrakcji w systemie"
          page="2"
          title="Atrakcje"
          onClickNavigate={mockOnClickNavigate}
        />
      </StyledComponentsProvider>
    );
    expect(paginationCounter).toHaveTextContent('2 / X');
  });
});
