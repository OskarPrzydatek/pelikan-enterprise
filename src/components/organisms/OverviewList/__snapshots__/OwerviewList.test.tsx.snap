// Vitest Snapshot v1

exports[`OverviewList > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ilhmMj jlDDjG"
      >
        <h2
          class="sc-dkrFOg kaKqDq"
        >
          Atrakcje
        </h2>
        <ul
          class="sc-jIRcFI cCkcaH"
        >
          <li
            class="sc-eDWCr jtwhuZ"
          >
            <p
              class="sc-iBYQkv cMzfYw"
            >
              attraction1
            </p>
            <div
              class="sc-bqWxrE gdYrVu"
            >
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
          >
            <p
              class="sc-iBYQkv cMzfYw"
            >
              attraction2
            </p>
            <div
              class="sc-bqWxrE gdYrVu"
            >
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
          >
            <p
              class="sc-iBYQkv cMzfYw"
            >
              attraction3
            </p>
            <div
              class="sc-bqWxrE gdYrVu"
            >
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
          >
            <p
              class="sc-iBYQkv cMzfYw"
            >
              attraction4
            </p>
            <div
              class="sc-bqWxrE gdYrVu"
            >
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
          <li
            class="sc-eDWCr jtwhuZ"
          >
            <p
              class="sc-iBYQkv cMzfYw"
            >
              attraction5
            </p>
            <div
              class="sc-bqWxrE gdYrVu"
            >
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="edit-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="22.5461"
                    stroke="#013059"
                    stroke-width="3"
                    width="21.0433"
                    x="2.05469"
                    y="7.50781"
                  />
                  <rect
                    fill="white"
                    height="12.0217"
                    width="12.0217"
                    x="15.5781"
                    y="2.99219"
                  />
                  <path
                    d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                    fill="#013059"
                  />
                  <rect
                    fill="#013059"
                    height="2.59198"
                    transform="rotate(-135 29.7031 5.03906)"
                    width="4.53906"
                    x="29.7031"
                    y="5.03906"
                  />
                </svg>
              </button>
              <button
                class="sc-hLBbgP dKSYeL"
              >
                <svg
                  data-testid="delete-icon"
                  fill="none"
                  height="20"
                  viewBox="0 0 22 33"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                    stroke="#F23557"
                    stroke-width="3"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="2"
                    x1="0.226562"
                    x2="21.4318"
                    y1="3.72656"
                    y2="3.72656"
                  />
                  <line
                    stroke="#F23557"
                    stroke-width="3"
                    x1="7.29688"
                    x2="14.3653"
                    y1="1.82031"
                    y2="1.82031"
                  />
                </svg>
              </button>
            </div>
          </li>
        </ul>
        <div
          class="sc-hhOBVt fKvNNb"
        >
          <button
            class="sc-hLBbgP ySCXl"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
          <p
            class="sc-iBYQkv cMzfYw"
            data-testid="pagination-counter"
          >
             / 
            5
          </p>
          <button
            class="sc-hLBbgP fksuwN"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
        </div>
        <button
          class="sc-gswNZR ebKtrE"
          data-testid="overview-list-navigate-to-create-page"
        >
          Dodaj atrakcję
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ilhmMj jlDDjG"
    >
      <h2
        class="sc-dkrFOg kaKqDq"
      >
        Atrakcje
      </h2>
      <ul
        class="sc-jIRcFI cCkcaH"
      >
        <li
          class="sc-eDWCr jtwhuZ"
        >
          <p
            class="sc-iBYQkv cMzfYw"
          >
            attraction1
          </p>
          <div
            class="sc-bqWxrE gdYrVu"
          >
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="edit-icon"
                fill="none"
                height="20"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="22.5461"
                  stroke="#013059"
                  stroke-width="3"
                  width="21.0433"
                  x="2.05469"
                  y="7.50781"
                />
                <rect
                  fill="white"
                  height="12.0217"
                  width="12.0217"
                  x="15.5781"
                  y="2.99219"
                />
                <path
                  d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                  fill="#013059"
                />
                <rect
                  fill="#013059"
                  height="2.59198"
                  transform="rotate(-135 29.7031 5.03906)"
                  width="4.53906"
                  x="29.7031"
                  y="5.03906"
                />
              </svg>
            </button>
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="delete-icon"
                fill="none"
                height="20"
                viewBox="0 0 22 33"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                  stroke="#F23557"
                  stroke-width="3"
                />
                <line
                  stroke="#F23557"
                  stroke-width="2"
                  x1="0.226562"
                  x2="21.4318"
                  y1="3.72656"
                  y2="3.72656"
                />
                <line
                  stroke="#F23557"
                  stroke-width="3"
                  x1="7.29688"
                  x2="14.3653"
                  y1="1.82031"
                  y2="1.82031"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          class="sc-eDWCr jtwhuZ"
        >
          <p
            class="sc-iBYQkv cMzfYw"
          >
            attraction2
          </p>
          <div
            class="sc-bqWxrE gdYrVu"
          >
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="edit-icon"
                fill="none"
                height="20"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="22.5461"
                  stroke="#013059"
                  stroke-width="3"
                  width="21.0433"
                  x="2.05469"
                  y="7.50781"
                />
                <rect
                  fill="white"
                  height="12.0217"
                  width="12.0217"
                  x="15.5781"
                  y="2.99219"
                />
                <path
                  d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                  fill="#013059"
                />
                <rect
                  fill="#013059"
                  height="2.59198"
                  transform="rotate(-135 29.7031 5.03906)"
                  width="4.53906"
                  x="29.7031"
                  y="5.03906"
                />
              </svg>
            </button>
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="delete-icon"
                fill="none"
                height="20"
                viewBox="0 0 22 33"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                  stroke="#F23557"
                  stroke-width="3"
                />
                <line
                  stroke="#F23557"
                  stroke-width="2"
                  x1="0.226562"
                  x2="21.4318"
                  y1="3.72656"
                  y2="3.72656"
                />
                <line
                  stroke="#F23557"
                  stroke-width="3"
                  x1="7.29688"
                  x2="14.3653"
                  y1="1.82031"
                  y2="1.82031"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          class="sc-eDWCr jtwhuZ"
        >
          <p
            class="sc-iBYQkv cMzfYw"
          >
            attraction3
          </p>
          <div
            class="sc-bqWxrE gdYrVu"
          >
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="edit-icon"
                fill="none"
                height="20"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="22.5461"
                  stroke="#013059"
                  stroke-width="3"
                  width="21.0433"
                  x="2.05469"
                  y="7.50781"
                />
                <rect
                  fill="white"
                  height="12.0217"
                  width="12.0217"
                  x="15.5781"
                  y="2.99219"
                />
                <path
                  d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                  fill="#013059"
                />
                <rect
                  fill="#013059"
                  height="2.59198"
                  transform="rotate(-135 29.7031 5.03906)"
                  width="4.53906"
                  x="29.7031"
                  y="5.03906"
                />
              </svg>
            </button>
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="delete-icon"
                fill="none"
                height="20"
                viewBox="0 0 22 33"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                  stroke="#F23557"
                  stroke-width="3"
                />
                <line
                  stroke="#F23557"
                  stroke-width="2"
                  x1="0.226562"
                  x2="21.4318"
                  y1="3.72656"
                  y2="3.72656"
                />
                <line
                  stroke="#F23557"
                  stroke-width="3"
                  x1="7.29688"
                  x2="14.3653"
                  y1="1.82031"
                  y2="1.82031"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          class="sc-eDWCr jtwhuZ"
        >
          <p
            class="sc-iBYQkv cMzfYw"
          >
            attraction4
          </p>
          <div
            class="sc-bqWxrE gdYrVu"
          >
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="edit-icon"
                fill="none"
                height="20"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="22.5461"
                  stroke="#013059"
                  stroke-width="3"
                  width="21.0433"
                  x="2.05469"
                  y="7.50781"
                />
                <rect
                  fill="white"
                  height="12.0217"
                  width="12.0217"
                  x="15.5781"
                  y="2.99219"
                />
                <path
                  d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                  fill="#013059"
                />
                <rect
                  fill="#013059"
                  height="2.59198"
                  transform="rotate(-135 29.7031 5.03906)"
                  width="4.53906"
                  x="29.7031"
                  y="5.03906"
                />
              </svg>
            </button>
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="delete-icon"
                fill="none"
                height="20"
                viewBox="0 0 22 33"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                  stroke="#F23557"
                  stroke-width="3"
                />
                <line
                  stroke="#F23557"
                  stroke-width="2"
                  x1="0.226562"
                  x2="21.4318"
                  y1="3.72656"
                  y2="3.72656"
                />
                <line
                  stroke="#F23557"
                  stroke-width="3"
                  x1="7.29688"
                  x2="14.3653"
                  y1="1.82031"
                  y2="1.82031"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          class="sc-eDWCr jtwhuZ"
        >
          <p
            class="sc-iBYQkv cMzfYw"
          >
            attraction5
          </p>
          <div
            class="sc-bqWxrE gdYrVu"
          >
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="edit-icon"
                fill="none"
                height="20"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="22.5461"
                  stroke="#013059"
                  stroke-width="3"
                  width="21.0433"
                  x="2.05469"
                  y="7.50781"
                />
                <rect
                  fill="white"
                  height="12.0217"
                  width="12.0217"
                  x="15.5781"
                  y="2.99219"
                />
                <path
                  d="M15.9013 18.8409L12.6971 18.8456L12.7018 15.6414L25.6516 2.69162L28.8511 5.89109L15.9013 18.8409Z"
                  fill="#013059"
                />
                <rect
                  fill="#013059"
                  height="2.59198"
                  transform="rotate(-135 29.7031 5.03906)"
                  width="4.53906"
                  x="29.7031"
                  y="5.03906"
                />
              </svg>
            </button>
            <button
              class="sc-hLBbgP dKSYeL"
            >
              <svg
                data-testid="delete-icon"
                fill="none"
                height="20"
                viewBox="0 0 22 33"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.40432 31.5006L1.89274 7.64062H19.7656L17.254 31.5006H4.40432Z"
                  stroke="#F23557"
                  stroke-width="3"
                />
                <line
                  stroke="#F23557"
                  stroke-width="2"
                  x1="0.226562"
                  x2="21.4318"
                  y1="3.72656"
                  y2="3.72656"
                />
                <line
                  stroke="#F23557"
                  stroke-width="3"
                  x1="7.29688"
                  x2="14.3653"
                  y1="1.82031"
                  y2="1.82031"
                />
              </svg>
            </button>
          </div>
        </li>
      </ul>
      <div
        class="sc-hhOBVt fKvNNb"
      >
        <button
          class="sc-hLBbgP ySCXl"
        >
          <svg
            data-testid="arrow-icon"
            fill="none"
            viewBox="0 0 40 26"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 13L39.75 0.00961876V25.9904L0 13Z"
              fill="#013059"
            />
          </svg>
        </button>
        <p
          class="sc-iBYQkv cMzfYw"
          data-testid="pagination-counter"
        >
           / 
          5
        </p>
        <button
          class="sc-hLBbgP fksuwN"
        >
          <svg
            data-testid="arrow-icon"
            fill="none"
            viewBox="0 0 40 26"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 13L39.75 0.00961876V25.9904L0 13Z"
              fill="#013059"
            />
          </svg>
        </button>
      </div>
      <button
        class="sc-gswNZR ebKtrE"
        data-testid="overview-list-navigate-to-create-page"
      >
        Dodaj atrakcję
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
